### PR TITLE
Collections

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #8214](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8214/artifact/build/artifact/wire-dev-PR460-8214.apk)
[Download build #8217](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8217/artifact/build/artifact/wire-dev-PR460-8217.apk)
[Download build #8218](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8218/artifact/build/artifact/wire-dev-PR460-8218.apk)
[Download build #8221](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8221/artifact/build/artifact/wire-dev-PR460-8221.apk)
[Download build #8222](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8222/artifact/build/artifact/wire-dev-PR460-8222.apk)
[Download build #8224](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8224/artifact/build/artifact/wire-dev-PR460-8224.apk)
[Download build #8226](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8226/artifact/build/artifact/wire-dev-PR460-8226.apk)
[Download build #8227](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8227/artifact/build/artifact/wire-dev-PR460-8227.apk)
[Download build #8238](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8238/artifact/build/artifact/wire-dev-PR460-8238.apk)
[Download build #8241](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8241/artifact/build/artifact/wire-dev-PR460-8241.apk)
[Download build #8249](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8249/artifact/build/artifact/wire-dev-PR460-8249.apk)
[Download build #8255](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8255/artifact/build/artifact/wire-dev-PR460-8255.apk)
[Download build #8261](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8261/artifact/build/artifact/wire-dev-PR460-8261.apk)
[Download build #8268](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8268/artifact/build/artifact/wire-dev-PR460-8268.apk)
[Download build #8269](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8269/artifact/build/artifact/wire-dev-PR460-8269.apk)

### CODE REVIEW

**Comment by JPonte** [comment]
build again

**Comment by xuanzhaopeng** [comment]
https://wearezeta.atlassian.net/browse/AN-4864

**Comment by JPonte** [comment]
build again

File: app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    val emptyView: View = ViewUtils.getView(view, R.id.ll__collection__empty)
+    val toolbar: Toolbar = ViewUtils.getView(view, R.id.t_toolbar)
+    emptyView.setVisibility(View.GONE)
+
+    controller.focusedItem.on(Threading.Ui) {
+      case Some(md) if md.msgType == Message.Type.ASSET => showSingleImage(md.assetId)
+      case _ => closeSingleImage()
+    }
+
+    val columns = 4
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala]: Maybe move `columns` into companion object to make it more visible and use it directly in the adapter.

File: app/src/main/scala/com/waz/zclient/conversation/CollectionRecyclerView.scala [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    val y = Math.round(event.getY)
+    val shouldIntercept = collectionItemDecorator.getHeaderClicked(x, y) >= 0
+    superIntercept || shouldIntercept
+  }
+}
+
+object CollectionRecyclerView {
+
+  val MaxSmoothScroll = 50
+  val columns = 4
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionRecyclerView.scala]: Maybe get this from the fragment so we don't risk getting out of sync?

File: app/src/main/res/layout/collection_message_file_asset_content.xml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        android:clipToPadding="false"
+        android:clipChildren="false"
+        >
+
+        <com.waz.zclient.ui.text.TypefaceTextView
+            android:id="@+id/ttv__collection_item__user_name"
+            android:layout_width="wrap_content"
+            android:layout_height="@dimen/collections__small_text_height"
+            android:gravity="bottom"
+            android:layout_gravity="center_vertical|left"
```
Comment (hakonbo) [inline app/src/main/res/layout/collection_message_file_asset_content.xml]: How does this work in right-to-left languages. Perhaps consider using `start|end` instead? See there are a few more cases of this in the XMLs.

**Pull request merged**